homeassistant/components/satel_integra/alarm_control_panel.py

```diff
@@ -42,10 +50,13 @@ def __init__(self, name, arm_home_mode, partition_id):
         self._state = None
         self._arm_home_mode = arm_home_mode
         self._partition_id = partition_id
+        self._satel = None
+
 
     async def async_added_to_hass(self):
         """Update alarm status and register callbacks for future updates."""
         _LOGGER.debug("Starts listening for panel messages")
+        self._satel = self.hass.data[DATA_SATEL]
```

Comment:
Pass the controller as an argument to init method instead. If we can, we should always set up the instance completely in the init method.

Reply:
Ah, now I get it.